pkg/types/spec/api.go

```diff
@@ -72,6 +74,40 @@ func GetAPISpec(apiConfig *userconfig.API, projectID string, deploymentID string
 	}
 }
 
+func (api *API) ModelIDs() []string {
+	models := []string{}
+	if api != nil && len(api.LocalModelCaches) > 0 {
+		for _, localModelCache := range api.LocalModelCaches {
+			models = append(models, localModelCache.ID)
+		}
+	}
+
+	return models
+}
+
+func (api *API) ModelNames() []string {
+	names := []string{}
+	if api != nil && len(api.Predictor.Models) > 0 {
+		for _, model := range api.Predictor.Models {
+			names = append(names, model.Name)
+		}
+	}
+
+	return names
+}
+
+func (api *API) SubtractModelIDs(apis ...*API) []string {
+	modelIDs := strset.FromSlice(api.ModelIDs())
+	for _, a := range apis {
+		modelIDs.Subtract(strset.FromSlice(a.ModelIDs()))
```

Comment:
nit: I think this would work and would be a little simpler:

```go
modelIDs.Remove(a.ModelIDs()...)
```

Reply:
Yep, that's a nice alternative. Gonna change this now.

Reply:
This is now done!